tests/storage/blockvolume_test.py

```diff
@@ -51,9 +53,43 @@
 ])
 
 
+@contextmanager
+def make_volume(size, format=sc.RAW_FORMAT, prealloc=sc.SPARSE_VOL):
+    img_uuid = make_uuid()
+    vol_uuid = make_uuid()
+    # TODO: fix make_volume helper to create the qcow image when needed
+    with fake_env(storage_type='block') as env:
+        if format == sc.RAW_FORMAT:
+            env.make_volume(
+                size, img_uuid, vol_uuid, vol_format=format, prealloc=prealloc)
+            vol = env.sd_manifest.produceVolume(img_uuid, vol_uuid)
+            yield vol
+        else:
+            chain = make_qemu_chain(
+                env, size, format, chain_len=1, prealloc=prealloc)
+            yield chain[0]
+
+
 @expandPermutations
 class TestBlockVolumeSize(VdsmTestCase):
 
+    @pytest.mark.xfail(reason='Vdsm over-allocates one chunk', strict=False)
+    @permutations([
+        # capacity
+        [1 * MiB],
+        [512 * MiB],
+        [sc.VG_EXTENT_SIZE],
+        [CHUNK_SIZE_MB * MiB - sc.VG_EXTENT_SIZE],
+        [(CHUNK_SIZE_MB + 1) * MiB - sc.VG_EXTENT_SIZE],
+        [(CHUNK_SIZE_MB - 1) * MiB],
+    ])
+    @MonkeyPatch(blockVolume, 'config', CONFIG)
+    def test_cow_volume_small_size(self, capacity):
+        with make_volume(capacity, format=sc.COW_FORMAT) as vol:
+            self.assertEqual(vol.getCapacity(), capacity)
+            initial_size = utils.round(capacity, sc.VG_EXTENT_SIZE)
+            self.assertEqual(vol.getVolumeSize(), initial_size)
```

Comment:
This repeats the blocksd tests in a bad way. I would delete the duplicate tests here instead of wasting time on the fake storage.

Reply:
Removed.